pkg/dhclient/dhclient.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2013-2019 the Webboot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dhclient
+
+import (
+	"context"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/u-root/u-root/pkg/dhclient"
+	"github.com/vishvananda/netlink"
+)
+
+// ConfigureAll sets up the dhcp configurations for all of the ifNames. 
+func ConfigureAll(ifName string, timeout int, retry int, verbose bool, ipv4 bool, ipv6 bool) {
+	ifRE := regexp.MustCompilePOSIX(ifName)
+
+	ifnames, err := netlink.LinkList()
+	if err != nil {
+		log.Fatalf("Can't get list of link names: %v", err)
+	}
+
+	var filteredIfs []netlink.Link
+	for _, iface := range ifnames {
+		if ifRE.MatchString(iface.Attrs().Name) {
+			filteredIfs = append(filteredIfs, iface)
+		}
+	}
+
+	if len(filteredIfs) == 0 {
+		log.Fatalf("No interfaces match %s", ifName)
+	}
+
+	channel := make(chan error)
+	go configureAlls(filteredIfs, channel, ifName, timeout, retry, verbose, ipv4, ipv6)
+	x := <-channel
+	log.Printf("The channel returns: %v", x)
+}
+
+func configureAlls(ifs []netlink.Link, channel chan error, ifName string, timeout int, retry int, verbose bool, ipv4 bool, ipv6 bool) {
```

Comment:
what is the purpose of this name ... configureAlls? What is the Alls here? All is already plural.

Reply:
this function is that looks for ipv4, ipv6, retries it 5 times, and times out with each retry 15 seconds added to it. It sends request to get connected. Once it finds and configures the interface with IPv4 or IPv6. It will print that. If it cannot find IPv4 or IPv6, then it will let us know that as well.

Reply:
Maybe, a good name for the internal function should remain to be configureAll without the s. However, let me try to come up with a better exported name :) !